### ISSUE TITLE
Remove hard dependencies between CDK Stacks in aws-backend

### ISSUE THREAD

**Issue by richerve** [your-digital-rights/yourdigitalrights.org#232]
Currently the backend code is using the AWS CDK and values are passed between the different Stacks forming a hard dependency between them. So to deploy the Stack that deploys the lambda function, the DBStack that deploys DynamoDB must be evaluated too.

In general this is not bad practice for a couple of reasons:
- If the DB changes that will most likely affect code that uses the DB, like the lambda function, so it makes sense to apply changes in chain
- Name and/or Id values autogenerated by the CDK can be passed automatically to other stacks that would need it, like the SNS queue name to the lambda function.

Although those benefits are there, having tight dependencies introduce issues when trying to test the Stacks independently or simply apply a change to one section of the infrastructure without affecting others.

There can still be stacks per components, but the values to connect them should be specified manually instead of being computed at runtime

**Comment by richerve** [comment]
Removing hard dependencies between stacks is not really feasible in this project, because almost every component depends on other.

- The lambda function stack depends on having the Tables defined in the DBStack both to connect to those and to grant IAM permissions so the lambda function can act on it. Passing only a DB Name is not enough
- The lambda function stack also depends on the IncomingEmailStack because the SNS queue is created there, and that is used to later subscribe the lambda function to it.

So, it makes sense that both the DBStack and InboundEmailStack can be tested and deployed independently but it will always affect the ConfirmationEmailStack that contains the lambda function, so this last one can't be decoupled from prior two.

**Comment by richerve** [comment]
I also tried to do this by creating a "linker" stack that would only be used to receive AWS objects as parameters and run commands that depends on them internally. Like linking the permissions for the lambda function to talk to the DB.

What I found with this approach is that the dependencies will stick exists between the stacks that originally create the objects, even if the link is defined elsewhere. It looks like when the CDK forms the set of objects after reading all the Stacks, the dependency graph is still formed from the original parents.
